app/services/subscription_reference_generator.rb

```diff
@@ -0,0 +1,32 @@
+class SubscriptionReferenceGenerator
+  def initialize(search_criteria:)
+    update(search_criteria)
+  end
+
+  def update(search_criteria)
+    @search_criteria = search_criteria
```

Comment:
Another point on instance variables, if you add `attr_reader :search_criteria`, you'll be able to access the instance variable as a standard variable, just another convention we use elsewhere, so would be good to be consistent.

Reply:
Ah. Good to know.